ext/RMagick/extconf.rb

```diff
@@ -186,6 +159,27 @@ def exit_failure(msg)
       exit(1)
     end
 
+    def determine_imagemagick_package
+      exit_failure "Can't install RMagick #{RMAGICK_VERS}. Can't find pkg-config in #{ENV['PATH']}\n" unless find_executable('pkg-config')
+
+      packages = `pkg-config --list-all`.scan(/(ImageMagick\-6[\.A-Z0-9]+) .*/)
+
+      if packages.length.zero?
+        exit_failure "Can't install RMagick #{RMAGICK_VERS}. Can't find ImageMagick with pkg-config\n"
+      end
+
+      if packages.length > 1
+        msg = "\nWarning: Found more than one ImageMagick installation. This could cause problems at runtime.\n"
+        packages.each do |package|
+          msg << " - #{package[0]}\n"
```

Comment:
This is mutating the string, which is something Ruby is gradually moving away from. Not sure what's the cleanest alternative, but maybe something like:

```rb
package_lines = packages.map { |package| " - #{package}" }.join("\n")
msg = "\nWarning...  \n#{package_lines}"
```

Reply:
Copied this from another code block but just changed it to how you suggested it.